### PR TITLE
Guides Dark Mode fixes

### PR DESCRIPTION
**Pull request by jathayde**
Screenshot 2024-03-28 at 9 38 25 PM

### Motivation / Background

This is a final (?) PR related to feedback on dark mode from various users and DHH/Core team members:

Referencing @claudioduarte in https://github.com/rails/rails/discussions/51415

- [x] Make alt, shorter SVG line for mobile, add responsive style
- [x] Fix padding on footer (goes to edge on mobile, does not align on desktop)
- [x] Shrink overall height of `subCol`/sidebar so it doesn’t overlap footer

Referencing @ivanjuric in https://github.com/rails/rails/discussions/51398

- [x] Flip inverted dark modes on “more Ruby on Rails” nav bar in mobile view

Notes from DHH & Core Team:

- [x] Darker dark mode (Github non-dimmed style)
- [x] Remove double box on WIP call out on index right side
- [x] NavBar bottom issue clipping on some guides (pad the content?)
- [x] Rounded corners/better appearance on select form elements in header
- [x] Black border on bottom of interstitials
- [x] Font size to 16px (100%) from 18px base
- [x] Brighten syntax highlighting across the board

### Additional information

Screenshot 2024-03-28 at 9 38 36 PM

Screenshot 2024-03-28 at 9 39 29 PM

### Checklist

Before submitting the PR make sure the following are checked:

* [x] This Pull Request is related to one change. Unrelated changes should be opened in separate PRs.
* [x] Commit message has a detailed description of what changed and why. If this PR fixes a related issue include it in the commit message. Ex: `[Fix #issue-number]`
* [x] Tests are added or updated if you fix a bug or add a feature.
* [x] CHANGELOG files are updated for the changed libraries if there is a behavior change or additional feature. Minor bug fixes and documentation changes should not be included.

### CODE REVIEW

**Comment by lloydk** [comment]
### To much contrast

I highly recommend lowering the contrast of the text in dark mode. To much contrast in dark mode can cause halation and makes the content hard to read. I think that lowering the contrast will result in a more pleasant reading experience.

[The Accessible Perceptual Contrast Algorithm (APCA)](https://github.com/Myndex) has some proposed guidelines for [maximum contrast for dark mode](https://github.com/Myndex/SAPC-APCA/discussions/106).

Proposed changes based on the APCA guidelines:

Text content should have a maximum contrast of Lc 90 and text >= 36px or bold text >= 24px (h1, h2, h3) should have a maximum contrast of Lc 75.

**Background of `#181818`**
- For normal text use `#e5e5e5` instead of `#ffffff`. This [results a contrast of Lc 89.9](https://www.myndex.com/apca/?BG=181818&TXT=e5e5e5&DEV=G4g&BUF=A22).
- For h1, h2, h3 use `#cccccc` instead of `#ffffff`. This [results in a contrast of 74.6](https://www.myndex.com/apca/?BG=181818&TXT=cccccc&DEV=G4g&BUF=A22).
 
**Background of `#292929` (Chapter guide)**
- Can use the same colors that were used for a background of `#181818`.

*Note that I'm not proposing any changes to the colors to the sample code.*

### Red/Pink links

I'll preface this by pointing out that [made the same changes](https://github.com/rails/rails/pull/42799) to pass the WCAG standard. 

**Readability should trump branding especially for documentation.**

I understand the desire to use red links for branding purposes and the red link color passes the WCAG standard (barely) but the links have a very low contrast compared to the rest of the text which is distracting and makes the content harder to read.

The red links have a very low APCA score Lc 44.6 which is very far from the minimum recommendation of Lc 75 for body text.

A random color that I picked `#94e0fa` which is Lc 80.1 or any color that gives an APCA Lc score between 75-90 will provide a better reading experience.

The hover color for the chapter links should also be changed to something more readable.

### Other suggestions

Change the background of `body.guide p code, ...` to `#303030` and the color to `#e5e5e5`. I think the difference between the background colors of `#181818` and `#505050` is distracting and a subtle background difference would be more readable. 

Lower the contrast and possibly change the color of the Chapters section to provide more "contrast" with the main content. APCA suggests a preferred contrast of Lc75 for primary navigation. For example the random blue I picked above (`#94e0fa`) would have an Lc of 77.6 against the background of `#292929`.


From [APCA Use Cases and Conformance Research](https://github.com/Myndex/SAPC-APCA/discussions/39)

> Fluent Readability, Primary content that is not body text
up to two lines of continuous text (per the normal non-zoomed layout)
No larger than an x-height of 80px (Verdana 144px or Times 178px)
No smaller than an x-height of 8px (Verdana 14.4px or Times 17.8px)
MINIMUM weight: 300 for fonts under 24px (x-height 12)
Minimum weight 200 for fonts above 24px.
MINIMUM contrast of Lc 60
PREFERRED contrast: Lc 75
includes headlines, captions, and images of text, if they contribute to content
primary navigation and primary menus
asides, tool-tips, spoilers, "continued"
user entered text in forms, fields, etc


I think this screenshot contains most of the suggested changes

image

**Comment by dhh** [comment]
I'm fine with the background color and text color from the comment above, but we'll stick to the red for the links. If there's slight tweak to that red that would improve things, I'm happy to consider that too. But we're not introducing a new off-brand color for this.

**Comment by lloydk** [comment]
I'll create a PR for the background color and text color changes. 

For the links how about something like this:

image

I've taken the logo red and lightened it until it passes APCA Lc 75 which is the minimum for body text and preferred APCA Lc for primary navigation.

**Comment by dhh** [comment]
Red in the sidebar is a no go. And the one in the main text looks rather sickly to me. But I'm good with background and the new white.

File: guides/assets/stylesheets/highlight.css
Comment (carlosantoniodasilva) [inline guides/assets/stylesheets/highlight.css:1]: @jathayde did you mean to commit the css files here again? They were being generated to guides/output when we first introduced them.

Comment (jathayde) [reply guides/assets/stylesheets/highlight.css:1]: Sorry, they should not be here.

Comment (carlosantoniodasilva) [reply guides/assets/stylesheets/highlight.css:1]: Ok thanks, I've pushed a commit removing them: 0c1804db40cd8fc027ebe61cf1ec9c6a8648605a

**Pull request merged**